stdlib/public/runtime/HeapObject.cpp

```diff
@@ -115,12 +118,77 @@ static HeapObject *_swift_tryRetain_(HeapObject *object)
     return _ ## name ## _ args; \
 } while(0)
 
+#if defined(_MALLOC_TYPE_ENABLED)
+static malloc_type_id_t computeTypeId(const HeapMetadata *heapMetadata) {
```

Comment:
I want to cache the `malloc_type_descriptor_t` (32 bit) value computed here.  Any suggestions on how to best do this?  We will require one value per heap-allocated type.

Reply:
The cost of caching it would probably exceed the cost of doing the computation here (which is going to compile to a couple of arithmetic instructions.)

I might factor it out into another function, but I don't think it's worth trying to cache unless profiling shows this computation is a bottleneck.